### DISCUSSION TITLE
Error while training,  INTERNAL ASSERT FAILED pytorch error

### DISCUSSION THREAD

**Discussion by Lonas13** [coqui-ai/TTS#2496] [General Q&A]
Hi, I am currently attempting to train my own dataset using the glowTTS model and formatting it according to the ljspeech dataset. Everything should be fine with my install as I've succesfully been able to generate text to speech and train the glowTTS model on the ljspeech dataset. However when I attempt to train my own dataset despite having it formatted identically (to the best of my ability) to the ljspeech dataset I keep running into this error when launching my training python file. 

`(ctts) lonas@DESKTOP-1C4LRGN:~/TTS/JDP/Data/JBP_Dataset$ CUDA_VISIBLE_DEVICES="0" python train_JBP.py
 > Setting up Audio Processor...
 | > sample_rate:22050
 | > resample:False
 | > num_mels:80
 | > log_func:np.log10
 | > min_level_db:-100
 | > frame_shift_ms:None
 | > frame_length_ms:None
 | > ref_level_db:20
 | > fft_size:1024
 | > power:1.5
 | > preemphasis:0.0
 | > griffin_lim_iters:60
 | > signal_norm:True
 | > symmetric_norm:True
 | > mel_fmin:0
 | > mel_fmax:None
 | > pitch_fmin:1.0
 | > pitch_fmax:640.0
 | > spec_gain:20.0
 | > stft_pad_mode:reflect
 | > max_norm:4.0
 | > clip_norm:True
 | > do_trim_silence:True
 | > trim_db:45
 | > do_sound_norm:False
 | > do_amp_to_db_linear:True
 | > do_amp_to_db_mel:True
 | > do_rms_norm:False
 | > db_level:None
 | > stats_path:None
 | > base:10
 | > hop_length:256
 | > win_length:1024
 | > Found 522 files in /home/lonas/TTS/JDP/Data/JBP_Dataset
 > Training Environment:
 | > Current device: 0
 | > Num. of GPUs: 1
 | > Num. of CPUs: 12
 | > Num. of Torch Threads: 6
 | > Torch seed: 54321
 | > Torch CUDNN: True
 | > Torch CUDNN deterministic: False
 | > Torch CUDNN benchmark: False
 > Start Tensorboard: tensorboard --logdir=/home/lonas/TTS/JDP/Data/JBP_Dataset/run-April-10-2023_11+00AM-c2d15cd4

 > Model has 28597969 parameters

 > EPOCH: 0/1000
 --> /home/lonas/TTS/JDP/Data/JBP_Dataset/run-April-10-2023_11+00AM-c2d15cd4

> DataLoader initialization
| > Tokenizer:
        | > add_blank: False
        | > use_eos_bos: False
        | > use_phonemes: False
| > Number of instances : 517
 | > Preprocessing samples
 | > Max text length: 1044
 | > Min text length: 25
 | > Avg text length: 154.02707930367504
 |
 | > Max audio length: 220522.0
 | > Min audio length: 220522.0
 | > Avg audio length: 220522.0
 | > Num. instances discarded samples: 0
 | > Batch group size: 0.

 > TRAINING (2023-04-10 11:00:19)

   --> STEP: 0/17 -- GLOBAL_STEP: 0
     | > current_lr: 0.00000
     | > step_time: 4.11560  (4.11555)
     | > loader_time: 2.07320  (2.07320)

> DataLoader initialization
| > Tokenizer:
        | > add_blank: False
        | > use_eos_bos: False
        | > use_phonemes: False
| > Number of instances : 5
 | > Preprocessing samples
 | > Max text length: 185
 | > Min text length: 113
 | > Avg text length: 147.6
 |
 | > Max audio length: 220522.0
 | > Min audio length: 220522.0
 | > Avg audio length: 220522.0
 | > Num. instances discarded samples: 0
 | > Batch group size: 0.

 > EVALUATION

 | > Synthesizing test sentences.

  --> EVAL PERFORMANCE
     | > avg_loader_time: 0.15839 (+0.00000)
     | > avg_loss: 3.94930 (+0.00000)
     | > avg_log_mle: 0.72161 (+0.00000)
     | > avg_loss_dur: 3.22769 (+0.00000)

 > BEST MODEL : /home/lonas/TTS/JDP/Data/JBP_Dataset/run-April-10-2023_11+00AM-c2d15cd4/best_model_17.pth

 > EPOCH: 1/1000
 --> /home/lonas/TTS/JDP/Data/JBP_Dataset/run-April-10-2023_11+00AM-c2d15cd4

> DataLoader initialization
| > Tokenizer:
        | > add_blank: False
        | > use_eos_bos: False
        | > use_phonemes: False
| > Number of instances : 517
 | > Preprocessing samples
 | > Max text length: 1044
 | > Min text length: 25
 | > Avg text length: 154.02707930367504
 |
 | > Max audio length: 220522.0
 | > Min audio length: 220522.0
 | > Avg audio length: 220522.0
 | > Num. instances discarded samples: 0
 | > Batch group size: 0.

 > TRAINING (2023-04-10 11:00:36)
/home/lonas/TTS/TTS/tts/layers/generic/wavenet.py:107: UserWarning: FALLBACK path has been taken inside: runCudaFusionGroup. This is an indication that codegen Failed for some reason.
To debug try disable codegen fallback path via setting the env variable `export PYTORCH_NVFUSER_DISABLE=fallback`
 (Triggered internally at ../third_party/nvfuser/csrc/manager.cpp:335.)
  acts = fused_add_tanh_sigmoid_multiply(x_in, g_l, n_channels_tensor)
 ! Run is kept in /home/lonas/TTS/JDP/Data/JBP_Dataset/run-April-10-2023_11+00AM-c2d15cd4
Traceback (most recent call last):
  File "/home/lonas/miniconda3/envs/ctts/lib/python3.8/site-packages/trainer/trainer.py", line 1591, in fit
    self._fit()
  File "/home/lonas/miniconda3/envs/ctts/lib/python3.8/site-packages/trainer/trainer.py", line 1544, in _fit
    self.train_epoch()
  File "/home/lonas/miniconda3/envs/ctts/lib/python3.8/site-packages/trainer/trainer.py", line 1309, in train_epoch
    _, _ = self.train_step(batch, batch_num_steps, cur_step, loader_start_time)
  File "/home/lonas/miniconda3/envs/ctts/lib/python3.8/site-packages/trainer/trainer.py", line 1141, in train_step
    outputs, loss_dict_new, step_time = self._optimize(
  File "/home/lonas/miniconda3/envs/ctts/lib/python3.8/site-packages/trainer/trainer.py", line 1025, in _optimize
    outputs, loss_dict = self._model_train_step(batch, model, criterion)
  File "/home/lonas/miniconda3/envs/ctts/lib/python3.8/site-packages/trainer/trainer.py", line 970, in _model_train_step
    return model.train_step(*input_args)
  File "/home/lonas/TTS/TTS/tts/models/glow_tts.py", line 407, in train_step
    outputs = self.forward(
  File "/home/lonas/TTS/TTS/tts/models/glow_tts.py", line 239, in forward
    z, logdet = self.decoder(y, y_mask, g=g, reverse=False)
  File "/home/lonas/miniconda3/envs/ctts/lib/python3.8/site-packages/torch/nn/modules/module.py", line 1501, in _call_impl
    return forward_call(*args, **kwargs)
  File "/home/lonas/TTS/TTS/tts/layers/glow_tts/decoder.py", line 131, in forward
    x, logdet = f(x, x_mask, g=g, reverse=reverse)
  File "/home/lonas/miniconda3/envs/ctts/lib/python3.8/site-packages/torch/nn/modules/module.py", line 1501, in _call_impl
    return forward_call(*args, **kwargs)
  File "/home/lonas/TTS/TTS/tts/layers/glow_tts/glow.py", line 213, in forward
    x = self.wn(x, x_mask, g)
  File "/home/lonas/miniconda3/envs/ctts/lib/python3.8/site-packages/torch/nn/modules/module.py", line 1501, in _call_impl
    return forward_call(*args, **kwargs)
  File "/home/lonas/TTS/TTS/tts/layers/generic/wavenet.py", line 107, in forward
    acts = fused_add_tanh_sigmoid_multiply(x_in, g_l, n_channels_tensor)
RuntimeError: The following operation failed in the TorchScript interpreter.
Traceback of TorchScript (most recent call last):
RuntimeError: The following operation failed in the TorchScript interpreter.
Traceback of TorchScript (most recent call last):
RuntimeError: stack.size() >= frames.back().function->n_inputs INTERNAL ASSERT FAILED at "../torch/csrc/jit/runtime/interpreter.cpp":241, please report a bug to PyTorch.`

I'm at a total loss as to what the issue could be other than maybe my dataset is somehow bad, but even then I've scoured it fairly hard and haven't come across anything wrong that I can see, but I also just don't know what to look for.

**Comment by p0p4k** [comment]
go to https://github.com/coqui-ai/TTS/blob/914280a556b06c957f48f18fb64b61b8b984b37d/TTS/tts/layers/generic/wavenet.py 
In that file remove `@torch.jit.script`. Try and report.

**Comment by siddas27** [comment]
@Lonas13 Did you fix it? I am getting the same error.

**Comment by erogol** [comment]
I've seen this happen when there is OOM
